### PR TITLE
[v3.6.0 pipeline] add pipeline macro

### PR DESCRIPTION
**Pull request by star-e**
added custom pipeline macro and enabled custom pipeline code path.

### Changelog

* added CUSTOM_PIPELINE_NAME to the macro.ts

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cocos/core/platform/macro.ts [hunk 1]
```diff
@@ -1071,6 +1071,17 @@ interface Macro {
...
     BATCHER2D_MEM_INCREMENT: number;
 
+    /**
+     * @zh 自定义渲染管线的名字（实验性）
+     * 引擎会根据名字创建对应的渲染管线（仅限Web平台）。如果名字为空，则不启用自定义渲染管线。
+     * 目前仅支持'Forward', 'Deferred'两种。
+     * @en The name of custom rendering pipeline (experimental)
+     * Engine will use the name to create the custom pipeline (Web only). If the name is empty, custom pipeline will be disabled.
+     * Currently only 'Forward' and 'Deferred' are supported.
+     * @default 144 KB
```
Comment (yanjifa) [inline cocos/core/platform/macro.ts:1081]: @default 144 KB？

**Comment by star-e** [comment]
typo，fixed

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: cocos/core/pipeline/custom/web-pipeline.ts [hunk 2]
```diff
@@ -416,6 +416,17 @@ export class WebPipeline extends Pipeline {
         this._constantMacros = str;
     }
 
+    public get usesDeferredPipeline (): boolean {
+        return this._usesDeferredPipeline;
+    }
```
Comment (pandamicro) [inline cocos/core/pipeline/custom/web-pipeline.ts]: Please put properties with other properties

**Review by pandamicro — APPROVED** [review]

File: cocos/core/pipeline/custom/web-pipeline.ts (hunk 2 above)
Comment (star-e) [reply cocos/core/pipeline/custom/web-pipeline.ts]: fixed

**Pull request merged**